### PR TITLE
fix: change app context and related comps

### PR DESCRIPTION
**Pull request by songwongtp**
## Describe your changes
- change app context and related comps

### CODE REVIEW

File: src/lib/app-provider/hooks/useInternalNavigate.ts [hunk 1]
```diff
@@ -12,6 +15,7 @@ interface NavigationArgs {
 
 export const useInternalNavigate = () => {
   const router = useRouter();
+  const defaultChainId = SUPPORTED_CHAIN_IDS[0] ?? "";
```
Comment (poomthiti) [inline src/lib/app-provider/hooks/useInternalNavigate.ts:18]: ditto

**Review by poomthiti — CHANGES_REQUESTED** [review]

File: src/env.ts [hunk 2]
```diff
@@ -1,75 +1,7 @@
...
-    default:
-      return {
-        example: "" as HumanAddr,
-      };
-  }
-};
+export const SUPPORTED_CHAIN_IDS: string[] =
+  process.env.NEXT_PUBLIC_SUPPORTED_CHAIN_IDS?.split(",") ?? [""];
 
 export const FALLBACK_LCD_ENDPOINT: Record<string, string> = {
```
[the remark this answers is not in the document]
Comment (evilpeach) [reply src/env.ts:6]: We'll remove it in the upcoming PR.

File: src/lib/app-provider/hooks/useCelatoneApp.ts [hunk 3]
```diff
@@ -1,14 +1,5 @@
...
-  CelatoneHumanAddress,
-} from "types";
 
 export const useCelatoneApp = () => {
-  return useApp<
-    CelatoneContractAddress,
-    CelatoneHumanAddress,
-    CelatoneConstants
-  >();
+  return useApp();
```
[the remark this answers is not in the document]
Comment (evilpeach) [reply src/lib/app-provider/hooks/useCelatoneApp.ts:4]: Agreed!

File: src/pages/_app.tsx [hunk 4]
```diff
@@ -74,7 +61,6 @@ const MyApp = ({ Component, pageProps }: AppProps) => {
           chains={[...chains, terra2testnet]}
           assetLists={[...assets, terra2testnetAssets]}
           wallets={wallets}
-          endpointOptions={availableChainsEndpoints}
```
Comment (poomthiti) [inline src/pages/_app.tsx]: if we remove this, the wallet will use keplr rpc for sending transaction not from config

**Review by poomthiti — CHANGES_REQUESTED** [review]

**Review by poomthiti — APPROVED** [review]

**Review by evilpeach — APPROVED** [review]

**Pull request merged**